app/src/main/java/ch/epfl/cs311/wanderwave/ui/screens/ProfileViewOnlyScreen.kt

```diff
@@ -33,11 +39,27 @@ import ch.epfl.cs311.wanderwave.viewmodel.SongList
  * @last update 2.0
  */
 @Composable
-fun ProfileViewOnlyScreen(profile: Profile) {
+fun ProfileViewOnlyScreen(profileId: String, navigationActions: NavigationActions) {
+
+  val viewModel: ProfileViewModel = hiltViewModel()
+  LaunchedEffect(profileId) {
+    println(profileId)
+    if (profileId != null) {
+      viewModel.getProfileByID(profileId)
+    }
```

Comment:
Again, would be better to have some sort of fallback if the profile can't be fetched for some reason

Reply:
I have done something like this, let me know what you think about it:
```suggestion
      if (profileId != null) {
      viewModel.getProfileByID(profileId)
    }else {
        withContext(Dispatchers.Main) {
            navigationActions.navigateTo(Route.MAIN)
            Log.e("No profile found", "No beacons found for the given id")
        }
    }
```